src/net.h

```diff
@@ -570,8 +570,39 @@ class CNodeStats
 
 
 
-
+/** Transport protocol agnostic message container.
+ * Ideally it should only contain receive time, payload,
+ * command and size.
+ */
 class CNetMessage {
+public:
+    CDataStream m_recv;              // received message data
+    int64_t m_time;                  // time (in microseconds) of message receipt.
+    bool m_valid_netmagic;
```

Comment:
So next step would be to move these transport-protocol-specific fields in the Transport Deserializer ? If a `CNetMessage` doesn't have a valid_header it shouldn't succeed deserialization right ?

Reply:
Next step would be to further avoid transport-specific stuff in `ProcessMessages()`. I think  #15206 and #15197 are a good next step (would need rebase after this).

Reply:
Ok thanks, going to check them once there are rebased